### PR TITLE
zsh completion: Prevent functions from clobbering each other, &c.

### PR DESCRIPTION
**Pull request by okdana**
I've been keeping a mental list of issues with systemd's zsh completion functions for a while and decided i'd try to work on a few of them over my break.

This PR specifically relates to the following three concerns, which i felt were closely related, easy to address, and relatively unobjectionable:

* Several functions define very generically named helper functions (like `_images` and `_kernels`) which may collide with other completion functions. It's customary to prefix helper functions by the name of the main function to guard against this

* Several helper function names (like `_localectl_command`) use singular terms where, by convention, they should use plural

* Several functions unconditionally (re-)define helper functions every time they're called, making it impossible to trace them for debugging (because the 'tag' is lost when redefining), and disrupting the ability of end-users to override them

---

The following concerns are **not** addressed by this PR, but i thought i would list them here in case i never get around to working on them all (very possible, honestly) and/or a project member feels strongly that i shouldn't bother:

* Most functions use initial capital letters for their descriptions, and some even use full sentence punctuation; completion descriptions in zsh should not do this (e.g., it should be 'reverse output', not 'Reverse output')

* Several functions use vague/generic/incorrect descriptions, such as 'options' to describe sub-commands, and 'SEC' to describe an optarg specifying a number of seconds

* Several functions use unconventional plural forms in descriptions, as in the above 'options'

* Many functions use `_call_program` really strangely. It shouldn't hurt anything usually, but there are some edge cases where it will, and it's just confusing and unsightly in general

* Most functions lack even basic option exclusivity information, which leads to unfriendly behaviour (mainly offering options where they aren't useful)

* Most functions call `_arguments` without `-s -S`, which makes completing options irritating in some cases

* Several functions use slow and verbose constructs like `while read` loops where parameter expansion would suffice

* Several functions mix array and scalar contexts, unnecessarily `typeset -U` arrays, &c.

* Several functions fail to set specific tags for argument completions (seen especially in the use of `_describe` and `_message`), which disrupts the ability of end-users to customise their behaviour

* Several functions use unconventional singular tag names

* Many functions unnecessarily use `_message` to produce unconventional 'no match' messages

* Several functions define very simple helper functions that are only used one time and should probably be folded into the function that calls them

* Several functions leak parameters into the surrounding scope

* Several functions create unnecessary parameters

* Most local parameter names are unnecessarily prefixed by `_` (a convention usually reserved for functions that take parameter names as arguments or that pass parameters down to other functions)

* Many functions set descriptions through `_arguments` specs that are superfluous and misleading because they're overridden by another function

* Several functions call `_values` incorrectly, for example using `-s` instead of `-S`, and not giving a description (so that the first completion possibility is swallowed up)

* `_systemd-run` command completion doesn't work at all

* `_journalctl`'s `_list_fields` function doesn't behave very well wrt the handling of `=` in field/value pairs

* `_localectl` produces spurious 'error' messages for some commands

* `_networkctl` can print errors to the screen

* Several options for `udevadm trigger` are misdefined

* `_sd_unit_files` is weird; i think someone was confused about `_description`

* Code style is generally inconsistent and often hard to follow (extremely long lines, mixed brace styles, strange white space, &c.)

There are a bunch of other little things but it's mostly just like non-idiomatic code and unconventional behaviours.

For reference, this is zsh's official (but non-exhaustive) completion style guide: https://github.com/zsh-users/zsh/blob/master/Etc/completion-style-guide

Please let me know if further PRs for these would be unwelcome. Otherwise, if i do keep going, i'll refer back to this one

### CODE REVIEW

**Comment by keszybz** [comment]
> Please let me know if further PRs for these would be unwelcome. Otherwise, if i do keep going, i'll refer back to this one

Either separate PRs or more patches in this one is OK. Whatever you prefer.

**Comment by okdana** [comment]
Thank you. Rebased. Not sure when i'll get around to the other stuff, so i'll probably do separate PRs for them.
